programming/robot_api/index.md

```diff
@@ -3,126 +3,119 @@ layout: page
 
-~~~~~ python
-R.motor_board.motors[0].power = 0.5   # WILL work, if motor 0 exists
-R.motor_board.motors[1].power = -0.2  # WILL work, if motor 1 exists
-R.motor_board.motors.power = 0.42     # WON'T WORK
+* [kch]({{ site.baseurl }}/programming/leds)
+* [motor_board]({{ site.baseurl }}/programming/motors)
+* [power_board]({{ site.baseurl }}/programming/power)
+* [servo_board]({{ site.baseurl }}/programming/servos)
+* [arduino]({{ site.baseurl }}/programming/arduino/)
+* [camera]({{ site.baseurl }}/programming/vision/)
 
-# the above is similar to the situation for 'ruggeduinos' and 'servos'
-~~~~~
+The functions of each board are described on their respective pages.
 
-A number of examples in the documentation will assume you've instantiated the required `Robot` class and have called it `R`.
-From here in, if you see a `R.something`, the requirement of the `sr.robot3` import line and the instantiation of `Robot` as `R` is implicit.
 
-[Other Robot Attributes](#OtherRobotAttributes) {#OtherRobotAttributes}
-----------------------
+## Other Robot Attributes
 
-As well as the attributes listed above, the Robot class also has the following attributes, which you may find useful:
+As well as the attributes listed above, the `Robot` class also has the following attributes, which you may find useful:
 
 zone
-:    The number of the zone that the robot is associated with. Between `0` and `3`.
+:   The number of the scoring zone that the robot is associated with.
+    Between `0` and `3`.
+
+    This attribute is only available after the start button is pressed and will throw an error if accessed before.
+    See the [competition mode](./comp_mode) page for more information about this attribute.
 
 mode
 :   Whether the robot is running in competition mode.
-    When in a competition match, this will be `RobotMode.COMP`, and at all other times this will be `RobotMode.DEV`.
+    When in a competition match, this will be `COMP`, and at all other times this will be `DEV`.
+
+    This attribute is only available after the start button is pressed and will throw an error if accessed before.
+    See the [competition mode](./comp_mode) page for more information about this attribute.
 
     ~~~~~ python
     from sr.robot3 import *
 
-    R = Robot()
+    robot = Robot()
 
-    if R.mode == RobotMode.COMP:
-      print("This is the competition!")
+    if robot.mode == COMP:
+        print("This is the competition!")
+    elif robot.mode == DEV:
+        print("This is development")
     ~~~~~
 
 usbkey
-:   The path to the USB memory stick.
-    Your code is unzipped and run from a temporary directory, therefore files you create will be lost when the kit is turned off.
-    You can use this to easily read from and write to files on the stick itself.
-    Note that the USB memory stick is mounted synchronously, so any writes to it will block until complete and may slow down your code.
+:   The path to the USB stick.
```

Comment:
Nit: `usbkey` is a `Path`, not a `str`. Should we reference this here:

```suggestion
:   A [`Path`](https://docs.python.org/3/library/pathlib.html#basic-use) object containing the path to the USB stick.
```

Reply:
1e11100